core/src/main/java/com/linecorp/armeria/client/AbstractClientFactory.java

```diff
@@ -96,18 +98,20 @@
     protected final Scheme validateScheme(URI uri) {
         requireNonNull(uri, "uri");
 
-        final String scheme = uri.getScheme();
-        if (scheme == null) {
-            throw new IllegalArgumentException("URI with missing scheme: " + uri);
-        }
-
-        if (uri.getAuthority() == null) {
+        if (Objects.isNull(uri.getAuthority())) {
             throw new IllegalArgumentException("URI with missing authority: " + uri);
         }
 
-        final Optional<Scheme> parsedSchemeOpt = Scheme.tryParse(scheme);
+        final String scheme = uri.getScheme();
+        if (Objects.isNull(scheme)) {
+            throw new IllegalArgumentException("URI with missing scheme: " + uri);
+        }
+        Optional<Scheme> parsedSchemeOpt = Scheme.tryParse(scheme);
```

Comment:
Instead of implementing the fallback logic here, how about modifying `Scheme.tryParse()` and `Scheme.parse()` so they accept a scheme without `none+`?

Reply:
Done. I have also removed some duplicate logic from Scheme.parse(). Now Scheme.parse() invoked tryParse(), and returns the value.

Reply:
Thanks! Looking nicer.